xbmc/cores/VideoPlayer/VideoRenderers/OverlayRenderer.cpp

```diff
@@ -259,6 +264,32 @@ void CRenderer::SetStereoMode(const std::string &stereomode)
   m_stereomode = stereomode;
 }
 
+void CRenderer::SetSubtitleVerticalPosition(const int value, bool save)
+{
+  if (save)
+  {
+    RESOLUTION_INFO resInfo;
```

Comment:
I don't know how GraphicContext class can be refactored to grant a more clean access,
but i can give a comment to the last sentence,
currently `iSubtitles` store the subtitle position on each resolution because
if you change resolution the scale is different then you need different value of subs position pixels
(currently these are stored and automatically restored on res change)

if we think to store this value outside resinfo like fixed value, happen this e.g.:

if you set the position to 100px (from bottom) and you have screen at 1080p
when you switch at 4k the subs will be too much bottom screen.

otherwise i think we should manage the position like do Libass as percentage % instead of pixels then we can store a single %value that can be used across all resolutions

Reply:
@enen92 my vision was to make VideoPlayer indenpendent from the GUI component. If VP was indenpendent from the UI, you could even use multiple instances of VP. That would allow PiP or transcoding a video stream by rendering not to the display but to an encoder.
@CastagnaIT independent from the GUI does also mean not to use settings like globals. Settings are mainly relevant when rendering to the display.

Reply:
@FernetMenta can you provide an example/link of what you mean for global settings?

Reply:
Doing things like
```
const std::shared_ptr<CSettings> settings = CServiceBroker::GetSettingsComponent()->GetSettings();
```
all over the place is like using globals. Basically this creates a dependency to the UI / settings.

Have a look at AE https://github.com/xbmc/xbmc/blob/master/xbmc/cores/AudioEngine/Engines/ActiveAE/ActiveAE.cpp#L2635
It loads all settings into a local variable. The other code in AE doesn't make calls to the settings component but uses the local copy.

Reply:
ok understand the way, but map all settings in a variable as on ActiveAE `m_settings`, in my opinion it is not always a big deal and IMO as is now i find it a fragmentary code implementation

so each class have to:
-store all required settings locally
-keep in sync the local settings values in some way

Pro:
-less calls to GUI setting component but in any case there is dependency to settings component, although (should be) confined to one method
Cons:
-as is implemented now, make more code to maintain (if a setting will be changed you have to rework structs and variables of all customs `m_settings` variables of each class, maybe use common structs that can be maintained on one place?)
-more memory occupied with same data, since each class have to store same settings values

at least for OverlayRenderer no need to store all settings used since needs always fresh data

but if i have to take some ispiration from this i could store locally the settings more used only, so for: GetSubtitleVerticalMargin() / ConvertLibass() methods,
but then i need also to keep in sync these local settings stored, this can be done with "Observable Notify()" callback,
but as you specified in comment below is too much to put in the callback, then i need to delegate this update to the "Render()" it's all an mess of "loop"